ui/packages/ace/lib/format.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+'use strict';
+
+// Based on similar script in Facebook/React
+// https://github.com/facebook/react/blob/052a5f27f35f453ee33a075df81bddf9ced7300a/scripts/prettier/index.js
+
+const path = require('path');
+const color = require('ansi-colors');
+const glob = require('glob');
+const prettier = require('prettier');
+const fs = require('fs');
+const listChangedFiles = require('./listChangedFiles');
+
+module.exports = ({ args }) => {
+  const { write = false, modified = false } = args;
+  const changedFiles = modified ? listChangedFiles() : null;
+  let warnings = false;
+  let errors = false;
+
+  function warn(message) {
+    warnings = true;
+    console.error(color.yellow(`\n${message}\n=> has formatting issues`));
+  }
+
+  function error(message) {
+    errors = true;
+    console.error(color.red(`\n${message}`));
+  }
+
+  function tellUserHowToFormat() {
+    console.error(
+      '\n' +
+        color.red(
```

Comment:
❓ This red is pretty hard to read against powershell blue, is there a lighter shade of red we could use?

Reply:
Choose your color: https://github.com/doowb/ansi-colors